presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractArrayBlock.java

```diff
@@ -156,6 +156,22 @@ public Block getSingleValueBlock(int position)
                 newValues);
     }
 
+    @Override
+    public long getSingleValueLogicalSizeInBytes(int position)
+    {
+        checkReadablePosition(position);
+
```

Comment:
I will put a null check here (same for Map/Row Block):
```
if (isNull(position)) {
      return 0;
}
```

I understand the current implementation will return 0 also. But it's not immediate obvious when I first look at the code.